cleanupNoncurrentVersions.js

```diff
@@ -327,18 +343,27 @@ function _triggerDeletesOnEligibleObjects(bucket, versions, deleteMarkers,
                    deleteMarkers[matchingDeleteMarkerIndex].Key < version.Key) {
                 ++matchingDeleteMarkerIndex;
             }
-            if (matchingDeleteMarkerIndex < deleteMarkers.length &&
-                deleteMarkers[matchingDeleteMarkerIndex].Key === version.Key &&
-                deleteMarkers[matchingDeleteMarkerIndex].IsLatest === true) {
+            const isDeleted = matchingDeleteMarkerIndex < deleteMarkers.length
+                && deleteMarkers[matchingDeleteMarkerIndex].Key === version.Key
+                && deleteMarkers[matchingDeleteMarkerIndex].IsLatest === true;
+
+            const isEligible = isDeleted && _deleteMarkerIsEligible(
+                deleteMarkers[matchingDeleteMarkerIndex].LastModified);
+
+            if (isDeleted && isEligible) {
```

Comment:
Since `isEligible` implies `isDeleted` now:

```suggestion
            if (isEligible) {
```